apps/calendar/test/unit/views/single_day_test.js

```diff
@@ -157,290 +163,302 @@ suite('Views.SingleDay', function() {
+      assert.include(busy.className, 'md__event', '4: busy md__event');
+      assert.include(busy.className, 'is-allday', '4: busy is-allday');
+      assert.equal(
+        busy.style.borderColor,
+        'rgb(0, 255, 204)',
+        '4: busy border color'
+      );
+      assert.equal(
+        busy.style.backgroundColor,
+       'rgba(0, 255, 204, 0.2)',
+       '4: busy background color'
+      );
+      assert.include(
+        busy.getAttribute('aria-labelledby'),
+        makeAllDayEventID('location'),
+        '4: busy location label'
+      );
+      assert.include(
+        busy.getAttribute('aria-labelledby'),
+        makeAllDayEventID('title'),
+        '4: busy title label'
+      );
+      assert.include(
+        busy.getAttribute('aria-labelledby'),
+        makeAllDayEventID('description'),
+        '4: busy description label'
+      );
+
+      remote = data.allday[0].event.remote;
+
+      title = busy.querySelector('.md__event-title');
+      assert.equal(
+        title.textContent.trim(),
+        remote.title,
+        '4: title'
+      );
+      assert.equal(
+        title.id,
+        makeAllDayEventID('title'),
+        '4: title id'
+      );
+
+      location = busy.querySelector('.md__event-location');
+      assert.equal(
+        location.textContent.trim(),
+        remote.location,
+        '4: location'
+      );
+      assert.equal(
+        location.id,
+        makeAllDayEventID('location'),
+        '4: location id'
+      );
+
+      description = busy.querySelector('#' + makeAllDayEventID('description'));
+      assert.equal(
+        description.getAttribute('aria-hidden'),
+        'true',
+        '4: description aria-hidden'
+      );
+      assert.equal(
+        description.getAttribute('data-l10n-id'),
+        'event-multiple-day-duration',
+        '4: description data-l10n-id'
+      );
+      assert.equal(
+        description.dataset.l10nArgs,
+        '{"startDate":"Wednesday, July 23, 2014","startTime":"00:00",' +
+          '"endDate":"Thursday, July 24, 2014","endTime":"00:00"}',
+        '4: description data-l10n-id'
+      );
+
+      data = {
+        amount: 2,
+        basic: [
+          makeRecord('Lorem Ipsum', '5:00', '6:00'),
+          makeRecord('Maecennas', '6:00', '17:00', 1)
+        ],
+        allday: []
+      };
+      // we always send all the events for that day and redraw whole view
+      subject._observer.write(data);
+
+      // it's enough to check only the new event and the amount of busytimes
+      // inside each holder (just to ensure we are removing old elements)
+      busytimes = daysHolder.querySelectorAll('.md__event');
+      assert.lengthOf(busytimes, 2, '5: events');
+      assert.lengthOf(
+        alldaysHolder.querySelectorAll('.md__event'), 0, '5: all'
+      );
+      assert.equal(
+        busytimes[1].querySelector('.md__event-title').textContent.trim(),
+        data.basic[1].event.remote.title,
+        '5: event title'
+      );
+    });
```

Comment:
I did not change the tests, just the indentation (grouped into a `suite` because of `setup/teardown`).

test is 250LOC because it is testing a lot of things at once.. we couldn't do the same thing with a single assertion without comparing the full string (that was the old logic and we always had to update the tests at each change to the markup - including whitespaces).

TL;DR; tests are good enough as is. trying to break it into 30+ tests makes no sense in this case. (we are making sure that the markup updates every time the data changes), as separate tests it would require way more setup/teardown logic.